File: backend/src/main/java/group/esig/todo/domain/models/Tarefa.java
package group.esig.todo.domain.models;


import group.esig.todo.domain.enums.TarefaStatus;
import lombok.*;

import group.esig.todo.domain.enums.Prioridade;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Tarefa {

    private UUID id;
    private String titulo;
    private String descricao;
    private Usuario responsavel;
    private Prioridade prioridade;
    private TarefaStatus status;
    private Instant deadline;
}

File: frontend/src/app/core/models/Tarefa.ts
import Situacao from '../enums/Situacao';
import Prioridade from '../enums/Prioridade';
import Usuario from './Usuario';

interface Tarefa {
    id: string;
    titulo: string;
    descricao: string;
    responsavel: Usuario;
    prioridade: Prioridade;
    status: Situacao;
    deadline: string;
}

export function getInitialTarefa (): Tarefa {
    return {
        id: "",
        titulo: "",
        descricao: "",
        responsavel: {
            id: "",
            nome: "",
            email: ""
        },
        prioridade: Prioridade.MEDIA,
        status: Situacao.ABERTA,
        deadline: "",
    }
}

export default Tarefa;
